R/check-all.R

```diff
@@ -0,0 +1,260 @@
+    data$file_data[manifest_index][[1]],
+    required_cols = config::get("complete_columns")$manifest,
+    success_msg = "No columns are empty in the manifest",
+    fail_msg = "Some columns are empty in the manifest"
+  )
+  empty_cols_indiv <- check_cols_empty(
+    data$file_data[indiv_index][[1]],
+    required_cols = config::get("complete_columns")$individual,
+    success_msg = "No columns are empty in the individual metadata",
+    fail_msg = "Some columns are empty in the individual metadata"
+  )
+  empty_cols_biosp <- check_cols_empty(
+    data$file_data[biosp_index][[1]],
+    required_cols = config::get("complete_columns")$biospecimen,
+    success_msg = "No columns are empty in the biospecimen metadata",
+    fail_msg = "Some columns are empty in the biospecimen metadata"
+  )
+  empty_cols_assay <- check_cols_empty(
+    data$file_data[assay_index][[1]],
+    required_cols = config::get("complete_columns")$assay,
+    success_msg = "No columns are empty in the assay metadata",
+    fail_msg = "Some columns are empty in the assay metadata"
+  )
+
+  # Incomplete required columns produce failures -----------------------------
+  complete_cols_manifest <- check_cols_complete(
+    data$file_data[manifest_index][[1]],
+    required_cols = config::get("complete_columns")$manifest,
+    success_msg = "All required columns present are complete in the manifest", # nolint
+    fail_msg = "Some required columns are incomplete in the manifest"
+  )
+  complete_cols_indiv <- check_cols_complete(
+    data$file_data[indiv_index][[1]],
+    required_cols = config::get("complete_columns")$individual,
+    success_msg = "All required columns present are complete in the individual metadata", # nolint
+    fail_msg = "Some required columns are incomplete in the individual metadata" # nolint
+  )
+  complete_cols_biosp <- check_cols_complete(
+    data$file_data[biosp_index][[1]],
+    required_cols = config::get("complete_columns")$biospecimen,
+    success_msg = "All required columns present are complete in the biospecimen metadata", # nolint
+    fail_msg = "Some required columns are incomplete in the biospecimen metadata" # nolint
+  )
+  complete_cols_assay <- check_cols_complete(
+    data$file_data[assay_index][[1]],
+    required_cols = config::get("complete_columns")$assay,
+    success_msg = "All required columns present are complete in the assay metadata", # nolint
+    fail_msg = "Some required columns are incomplete in the assay metadata" # nolint
+  )
+
+  # Metadata files appear in manifest ----------------------------------------
+  meta_files_in_manifest <- check_files_manifest(
+    data$file_data[manifest_index][[1]],
+    c(
+      data$file_name[indiv_index],
+      data$file_name[biosp_index],
+      data$file_name[assay_index]
+    ),
+    success_msg = "Manifest file contains all metadata files",
+    fail_msg = "Manifest file does not contain all metadata files"
+  )
+
+  ## List results
+  res <- list(
```

Comment:
Maybe let's name the list elements so that in testing we can subset by name rather than index (seems less fragile that way?)

Reply:
Having a named list seems to work fine. Yay